PUBDEV-8674: expose GLM p-values, z-values, std-error to python and R clients

wendycwong opened the pull request:
This PR completes the task in JIRA: https://h2oai.atlassian.net/browse/PUBDEV-8674 .

I have added p_values(), z_values(), std_error() to both R and Python client so that they don't have to dig into the model to get those values.

I added R and Python unit tests to make sure they work.  Errors will be rasied if compute_p_values=FALSE or other models other than glm calls those functions.

meganjkurka commented:
@wendycwong I think as an end user I would prefer if I could see the results in one table (i.e. feature name, coefficient, p-value, z-value, std error).  What do you think?

I am just thinking that if I only get a list of p-values, I will have to then add a step to get the feature names that correspond and other information.

wendycwong commented:
@meganjkurka Currently, if you call model.p_values(), model.z_values(), model.std_error(), you will get a dict in Python like this:

<img width="1580" alt="image" src="https://user-images.githubusercontent.com/7231712/165637879-eb5e3480-18ff-4617-b4fe-6d96e4ff5a59.png">

In R, this is what you see if you call h2o.p_values(model), h2o.z_values(model), h2o.std_error(model) and print out the values:

<img width="875" alt="image" src="https://user-images.githubusercontent.com/7231712/165638363-d4212c88-58dd-41d1-ab4a-39ff23a57248.png">

wendycwong commented:
@meganjkurka I added the new function coef_with_p_values() for Python and if you print it, you will see:

<img width="767" alt="image" src="https://user-images.githubusercontent.com/7231712/165642264-ae7c6e59-f7e7-4471-a26e-ed70941db2f8.png">

For R, I added the new function h2o.coef_with_p_values(model) for R and if you print it, you will see:
<img width="588" alt="image" src="https://user-images.githubusercontent.com/7231712/165644070-71131564-7a9e-4246-bb53-c2617d43c328.png">

wendycwong commented:
@meganjkurka:  What do you think now?

michalkurka reviewed (approved):
Thank you @wendycwong!
